src/frontend/src/lib/utils/nfts.utils.ts

```diff
@@ -0,0 +1,34 @@
+import type { Erc721CustomToken } from '$eth/types/erc721-custom-token';
+import type { NetworkId } from '$lib/types/network';
+import type { Nft } from '$lib/types/nft';
+import { isNullish, nonNullish } from '@dfinity/utils';
+
+export type NftsByNetwork = Record<NetworkId, Record<string, Nft[]>>;
+
+export const getLoadedNftsByTokens = ({
+	tokens,
+	loadedNfts
+}: {
+	tokens: Erc721CustomToken[];
+	loadedNfts: Nft[];
+}): NftsByNetwork => {
+	const nftsByToken: NftsByNetwork = {};
+
+	tokens.forEach((token) => {
```

Comment:
maybe deconstruct for readability?

```suggestion
	tokens.forEach(({address, network:{id:networkId}) => {
```